ArduCopter/mode_follow.cpp

```diff
@@ -82,16 +82,18 @@ void Copter::ModeFollow::run()
             dir_to_target_xy.normalize();
         }
 
-        // slow down horizontally as we approach target (use 1/2 of maximum deceleration for gentle slow down)
-        const float dist_to_target_xy = Vector2f(dist_vec_offs_neu.x, dist_vec_offs_neu.y).length();
-        copter.avoid.limit_velocity(pos_control->get_pos_xy_p().kP().get(), pos_control->get_max_accel_xy() * 0.5f, desired_velocity_xy_cms, dir_to_target_xy, dist_to_target_xy, copter.G_Dt);
+        if (g2.follow.avoidance_enabled()) {
+            // slow down horizontally as we approach target (use 1/2 of maximum deceleration for gentle slow down)
+            const float dist_to_target_xy = Vector2f(dist_vec_offs_neu.x, dist_vec_offs_neu.y).length();
+            copter.avoid.limit_velocity(pos_control->get_pos_xy_p().kP().get(), pos_control->get_max_accel_xy() * 0.5f, desired_velocity_xy_cms, dir_to_target_xy, dist_to_target_xy, copter.G_Dt);
```

Comment:
I think the `if` should be closed here; doing that still makes Follow mode benefit from avoidance (if it is enabled and only for fences) but it won't avoid the vehicle if the parameter is set to disable.